liquid-integration/src/main/java/com/e/liquid_integration/ui/LiquidBuyAndSellDashActivity.kt

```diff
@@ -0,0 +1,458 @@
+import org.dash.wallet.common.data.ExchangeRate
+import org.dash.wallet.common.util.GenericUtils
+import org.json.JSONObject
+
+
+class LiquidBuyAndSellDashActivity : InteractionAwareActivity() {
+
+    companion object {
+        fun createIntent(context: Context?): Intent? {
+            return if (LiquidClient.getInstance()!!.isAuthenticated) {
+                Intent(context, LiquidBuyAndSellDashActivity::class.java)
+            } else {
+                Intent(context, LiquidSplashActivity::class.java)
+            }
+        }
+    }
+
+    private var liquidClient: LiquidClient? = null
+
+    private lateinit var _context: Context
+    private var loadingDialog: ProgressDialog? = null
+
+    private val cryptoCurrencyArrayList = ArrayList<PayloadItem>()
+    private val fiatCurrencyList = ArrayList<PayloadItem>()
+    private var isSelectFiatCurrency = false
+    private var isClickLogoutButton = false
+
+    // var currentExchangeRate: LiveData<ExchangeRate>? = null
+    var currentExchangeRate: ExchangeRate? = null
+
+    override
+    fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_liquid_buy_and_sell_dash)
+        this._context = this@LiquidBuyAndSellDashActivity
+        liquidClient = LiquidClient.getInstance()
+
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolbar)
+        val actionBar = supportActionBar
+        actionBar?.apply {
+            setDisplayHomeAsUpEnabled(true)
+            setDisplayShowHomeEnabled(true)
+        }
+        setTitle(getString(R.string.buy_sell_dash_))
+
+        loadingDialog = ProgressDialog(this)
+        loadingDialog!!.isIndeterminate = true
+        loadingDialog!!.setCancelable(false)
+        loadingDialog!!.setMessage(getString(R.string.loading))
+
+        findViewById<LinearLayout>(R.id.llDisconnect).setOnClickListener { openLogOutUrl() }
+        findViewById<LinearLayout>(R.id.buy_dash).setOnClickListener { buyDash() }
+
+        findViewById<LinearLayout>(R.id.sell_dash).setOnClickListener {
+            showSellDashDialog()
+        }
+
+        findViewById<LinearLayout>(R.id.llTransferToLiquid).setOnClickListener {
+
+            /* val intent = Intent()
+             intent.setClassName(this, "de.schildbach.wallet.ui.LiquidDashToDashTransferActivity")
+             intent.putExtra("extra_title", getString(R.string.liquid))
+             intent.putExtra("extra_message", "Enter the amount to transfer")
+             intent.putExtra("extra_max_amount", "17")
+             startActivityForResult(intent, 101)*/
+            val walletDataProvider = application as WalletDataProvider
+            val address = Address.fromBase58(TestNet3Params.get(), "yTgh4Z1RrMXbJrbkbS7Lgk8NEZERJigMsy")
+            val amount = Coin.CENT
+            walletDataProvider.startSendCoinsForResult(this, 1234, address, amount)
```

Comment:
This has a hardcoded address and hardcoded reference to the network (TestNet3Params).  Neither should be present.  Though I don't think `Constants.NETWORK_PARAMETERS` is available to this module.  If not then the current network parameters should be made available in the `WalletDataProvider` 

Is this class used for anything useful?

Reply:
This will be fixed in milestone 3, This is partially developed.